pyvista/core/utilities/geometric_sources.py

```diff
@@ -40,6 +42,59 @@
 DOUBLE_PRECISION = _vtk.vtkAlgorithm.DOUBLE_PRECISION
 
 
+@abstract_class
+class BaseSource(_vtk.vtkAlgorithm):
+    """Superclass for all sources.
+
+    Parameters
+    ----------
+    point_dtype : str, default: 'float32'
+        Set the desired output point types. It must be either 'float32' or 'float64'.
+    """
+
+    def __init__(self, point_dtype='float32'):
```

Comment:
Is the default `point_dtype`, `SINGLE_PRECISION` for all sources here?  It seems like in a cursory check, a lot of them are, but if there are any that are `DOUBLE_PRECISION` this choice would change the vtk default, perhaps with poor outcomes.  In that case, it might be better to make the default `None` and defer to the vtk default.

Another possibility is to find all the cases that have `DOUBLE_PRECISION` and override the `__init__` method.  This is more maintenance burden, but is more clean for the documentation and user perspective.